great_tables/gt.py

```diff
@@ -258,8 +255,21 @@ def __init__(
 
     # -----
 
+    def _repr_html_(self):
+        import os
+
+        # Check if we are rendering in the Quarto environment
+        quarto_enabled = os.environ.get("QUARTO_BIN_PATH") is not None
+
```

Comment:
It might be nice to do some handling to make a guess on if you should make a page or not. Eg, make_page should be false if in an IPython terminal or Jupyter Notebook, but true in other occasions?

You can do checks for these situations with something like:

```python
options = ["TerminalInteractiveShell", "ZMQInteractiveShell"]
make_page = ipython.get_ipython().__class__.__name__ not in options
```

where
 Jupyter Notebooks ➡️ 'ZMQInteractiveShell'
IPython in a terminal  ➡️ 'TerminalInteractiveShell

Reply:
The latest in this PR includes code to determine whether a full page should be returned. Thanks for the example code!

Reply:
Just a quick note, ZMQInteractiveShell is a subclass of TerminalInteractiveShell, so we can pick up either in a isinstance(..., TerminalInteractiveShell) check

Reply:
No wait, ZMQ subclass InteractiveShell, which is a lil bit different 😅

Reply:
For posterity, here is all the handy checks plotly does on environments:

https://github.com/plotly/plotly.py/blob/95cc3d0dc6b3379d181e28ba90a0b708fad3bc40/packages/python/plotly/plotly/io/_renderers.py#L514

Reply:
@isabelizimm I've consolidated logic for detecting environments here:

https://github.com/posit-dev/great-tables/pull/233/files#diff-4725a5793cb0f0916ca5679673d3a72a4f2896f336e87af59f5cacc157b04dfeR15

We can tack things onto there pretty easily!